IQSS/dataverse
# Adds new update dataverse general endpoint

**What this PR does / why we need it**:

Adds a new endpoint (`PUT /api/dataverses/<identifier>`) for updating an existing Dataverse collection using a JSON file following the same structure as the one used in the API for the creation.

**Which issue(s) this PR closes**:

- Closes #10904 

**Special notes for your reviewer**:

I have refactored commands `CreateDataverseCommand` and `UpdateDataverseCommand`, so that they now inherit from a common parent class called `AbstractWriteDataverseCommand`, where the shared logic of these commands is centralized.

**Suggestions on how to test this**:

- Create a Dataverse using JSF, or the UI
- Create a json file using https://github.com/IQSS/dataverse/blob/develop/doc/sphinx-guides/source/_static/api/dataverse-complete-optional-params.json as example. Include in the JSON file only the properties you want to update.

Call the new API endpoint using curl:

`curl -H "X-Dataverse-key:xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx" -X PUT "http://localhost:8080/api/dataverses/<ALIAS_OF_CREATED_DV>" --upload-file <YOUR_JSON_FILE>
`

Verify that the new properties specified in the JSON are correctly reflected in the dataverse collection.

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:

No.

**Is there a release notes update needed for this change?**:

Yes. Attached.
